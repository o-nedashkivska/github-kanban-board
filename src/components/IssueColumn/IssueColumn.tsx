import { Col, Typography, Flex } from "antd";
import IssueCard from "../IssueCard";

import styles from "./issues-column.module.css";

interface IssueColumnProps {
  title: string;
}

const { Title } = Typography;

const IssueColumn: React.FC<IssueColumnProps> = ({
  title,
}: IssueColumnProps) => {
  return (
    <Col span={6} className={styles.column}>
      <Title level={4} className={styles["column__title"]}>
        {title}
      </Title>
      <Flex
        gap="middle"
        align="center"
        vertical
        className={styles["column__content"]}
      >
        <IssueCard />
        <IssueCard />
        <IssueCard />
      </Flex>
    </Col>
  );
};

export default IssueColumn;
