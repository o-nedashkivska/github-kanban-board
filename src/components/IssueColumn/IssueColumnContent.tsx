import { useSelector } from "react-redux";
import { Typography } from "antd";
import { SyncOutlined } from "@ant-design/icons";
import IssueCard from "../IssueCard";
import { getCurrentIssuesStatus } from "../../store/selectors";

import styles from "./issues-column.module.css";

interface IssueColumnContentProps {
  columnName: string;
  issues: Array<any>;
}
const { Title } = Typography;

const IssueColumnContent: React.FC<IssueColumnContentProps> = ({
  columnName,
  issues,
}) => {
  const status = useSelector(getCurrentIssuesStatus);

  if (status[columnName] === "loading")
    return <SyncOutlined className={styles.loader} spin />;

  return (
    <>
      {issues.length === 0 ? (
        <Title level={4}>No issues</Title>
      ) : (
        issues.map((issue, index) => (
          <IssueCard key={issue.id} index={index} {...issue} />
        ))
      )}
    </>
  );
};

export default IssueColumnContent;
