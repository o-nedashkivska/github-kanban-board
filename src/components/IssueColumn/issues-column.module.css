.column {
  padding-bottom: 50px;
}

.column__title {
  text-align: center;
}

.column__content {
  background-color: #ced4da;
  border: solid 1px black;
  padding: 5% 0;
  height: calc(100% - 50px);
  position: relative;
}

.column__content::before {
  position: absolute;
  top: 1px;
  bottom: 1px;
  left: -3px;
  right: -3px;
  display: block;
  border: 1px solid black;
  content: "";
  transform: rotate(0.1deg);
}

.column__content::after {
  position: absolute;
  top: -2px;
  bottom: -2px;
  left: 1px;
  right: 1px;
  display: block;
  border: 1px solid black;
  content: "";
  transform: rotate(-0.3deg);
}
