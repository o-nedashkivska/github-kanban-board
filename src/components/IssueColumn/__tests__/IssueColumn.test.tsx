import { screen } from "@testing-library/react";

import { renderWithProviders } from "../../../test-utils";

import IssueColumn from "../IssueColumn";

describe("IssueColumn", () => {
  test("displays correct title", () => {
    const { asFragment } = renderWithProviders(
      <IssueColumn title="ToDo" columnName="toDo" issues={[]} />
    );

    expect(screen.getByText("ToDo")).toBeInTheDocument();
    expect(screen.getByText("No issues")).toBeInTheDocument();

    expect(asFragment()).toMatchSnapshot();
  });

  test("passes issues to child component", () => {
    const issues = [
      {
        title: "Title 1",
        index: 0,
        number: 1011,
        comments: 11,
        created_at: new Date(),
      },
      {
        title: "Title 2",
        index: 1,
        number: 2022,
        comments: 22,
        created_at: new Date(),
      },
    ];

    const { asFragment } = renderWithProviders(
      <IssueColumn title="ToDo" columnName="toDo" issues={issues} />
    );

    expect(screen.getByText("ToDo")).toBeInTheDocument();

    issues.forEach(({ title }) =>
      expect(screen.getByText(title)).toBeInTheDocument()
    );

    expect(asFragment()).toMatchSnapshot();
  });
});
