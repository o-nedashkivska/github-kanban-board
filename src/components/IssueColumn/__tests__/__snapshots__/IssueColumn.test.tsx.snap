// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IssueColumn displays correct title 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-6 column css-dev-only-do-not-override-1kuana8"
  >
    <h4
      class="ant-typography column__title css-dev-only-do-not-override-1kuana8"
    >
      ToDo
    </h4>
    <div
      class="column__content ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-middle ant-flex-vertical"
    >
      <h4
        class="ant-typography css-dev-only-do-not-override-1kuana8"
      >
        No issues
      </h4>
    </div>
  </div>
</DocumentFragment>
`;

exports[`IssueColumn passes issues to child component 1`] = `
<DocumentFragment>
  <div
    class="ant-col ant-col-6 column css-dev-only-do-not-override-1kuana8"
  >
    <h4
      class="ant-typography column__title css-dev-only-do-not-override-1kuana8"
    >
      ToDo
    </h4>
    <div
      class="column__content ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-center ant-flex-gap-middle ant-flex-vertical"
    >
      <div
        class="ant-card ant-card-bordered ant-card-small card css-dev-only-do-not-override-1kuana8"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-stretch ant-flex-vertical"
          >
            <span
              class="ant-typography css-dev-only-do-not-override-1kuana8"
            >
              Title 1
            </span>
            <span
              class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
            >
              #1011 opened today
            </span>
            <span
              class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
            >
              No assignee | Comments: 11
            </span>
          </div>
        </div>
      </div>
      <div
        class="ant-card ant-card-bordered ant-card-small card css-dev-only-do-not-override-1kuana8"
      >
        <div
          class="ant-card-body"
        >
          <div
            class="ant-flex css-dev-only-do-not-override-1kuana8 ant-flex-align-stretch ant-flex-vertical"
          >
            <span
              class="ant-typography css-dev-only-do-not-override-1kuana8"
            >
              Title 2
            </span>
            <span
              class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
            >
              #2022 opened today
            </span>
            <span
              class="ant-typography ant-typography-secondary css-dev-only-do-not-override-1kuana8"
            >
              No assignee | Comments: 22
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
